src/plugins/intel_gna/src/pre_post_process/avx2.cpp

```diff
@@ -0,0 +1,317 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#include "avx2.hpp"
+
+#ifdef HAVE_AVX2
+#    include <immintrin.h>
+
+#    include "preprocessing.hpp"
+#endif  // HAVE_AVX2
+
+namespace ov {
+namespace intel_gna {
+namespace pre_post_processing {
+
+#ifdef HAVE_AVX2
+void convert_matrix_fp32_to_int16_avx(int16_t* ptr_dst,
```

Comment:
what is more I would create interface this conversion methods 
```
class DataStoragesConverter{

virtual convert_matrix_fp32_to_int16_avx() = 0;
virtual convert_matrix_fp32_to_int8_avx() = 0;
}
```
and create factory to return implementation
```
class ConverterFactory{

std::shared_ptr<DataStoragesConverter> create_converter();
};
```
you could move all code the source file.

```
#ifdef HAVE_AVX2
#    include <immintrin.h>

#    include "data_converter_impl.hpp"
#endif  // HAVE_AVX2

ConverterFactory::create_converter(){

#ifdef HAVE_AVX2
    return std::make_shared<DataStoragesConverterImpl>();
#else
    return nullptr;
#endif  // HAVE_AVX2
}
```

in gna plugin you could use it as 
```

auto storage_converter = ConverterFactory::create_converter();
InputOutputDataHandlerImpl handler(storage_converter );
handler.import_frames();
handler.export_scores();
```

It would easy implementation of tests and clarify implementation( due the fact knowledge about if avx is enabled would be hidden in factory.) For tests you could emulate it or use proper implementation if available.